framework/src/base/MooseError.C

```diff
@@ -0,0 +1,26 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "MooseError.h"
+
+namespace Moose {
+
+void MPIAbortWrapper()
+{
+#if defined(LIBMESH_HAVE_MPI)
+  MPI_Abort(libMesh::GLOBAL_COMM_WORLD, 1);
+#endif
```

Comment:
Of course this occurred to me, too. :-/

Unfortunately clang then errors out because it thinks in a ton of functions the end of the body is reached without returning a value!